### PR TITLE
Do not use pkg_resources

### LINKED ISSUE #20: import of sdsstools is slow-ish
Importing of `sdsstools` takes 0.5 seconds.   This isn't necessarily slow but since we import it at the beginning of our packages, it can contribute to slow package imports.   From the screenshot you can see the main culprit is `pkg_resources`, which is notoriously slow.  I think this is in part due to the fact it scans the full environment upon import, which incurs a high cost for big environments.  The use in `metadata.py` of `pkg_resources` can be effectively replaced by switching to `importlib.metadata.version` (`importlib_metadata.version` for py<=3.7) for finding package versions, and `packaging.version.parse` for version compliance to PEP440.   This reduces the metadata import time significantly but doesn't necessarily decrease the overall package import due to the use of `pkg_resources` further down in `sdsstools._vendor.releases.docutils...`.  

I'm not necessarily suggesting we fix it at this time, but I'm making an issue for awareness and to keep track of it.   

sdsstools_importtimes

### PR DESCRIPTION
**Pull request by albireox**
Closes #20.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,6 +26,7 @@ invoke = "^1.3.0"
 semantic-version = "^2.8.4"  # For sdsstools.releases
 docutils = "^0.16"
 six = "^1.14.0"
+packaging = "^20.4"
```
Comment (havok2063) [inline pyproject.toml:30]: Should `importlib_metadata` be added to the list of dependencies?

**Review by havok2063 — APPROVED** [review]
@albireox This looks good to me.  Testing this, the import time has come down for me from 0.5 to 0.1 seconds.  

With moving `releases` out of `_vendor`, will this change how products using the `sdsstools` version of `releases` need to import it?

File: pyproject.toml (hunk 1 above)
Comment (albireox) [reply pyproject.toml:31]: I added it for Python <3.7. It must have been installed by some other package, or at least the tests for Python 3.6 in the workflow didn't fail, but it's now added explicitely.

**Comment by albireox** [comment]
No, it should work the same as far as I've tested it. What the old code did was to expose `sdsstools._vendor.releases` as if it were `sdsstools.releases`, but to do that it needed to import it. Now it's doing that but without any dark magic and without need for an import on `__init__.py`.

**Pull request merged**